x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -309,12 +308,6 @@ private void createApiKeyAndIndexIt(
         Set<RoleDescriptor> userRoleDescriptors,
         ActionListener<CreateApiKeyResponse> listener
     ) {
-        if (TcpTransport.isUntrustedRemoteClusterEnabled()
```

Comment:
I wouldn't mind making this change in a separate PR: we can start with just the generic changes to `RBACEngine.getRemoteAccessRole...` and the Role changes, then a separate update around enabling this for API keys (with all the API key specific tests). I don't feel strongly though since the scope of the change isn't huge.